src/design-pages/passwordless_gdm.rst

```diff
@@ -0,0 +1,461 @@
+            "name": "Web Login",
+            "role": "eidp",
+            "selectable": true,
+            "init_prompt": "Login",
+            "link_prompt": "Login online with another device",
+            "uri": "https://short.url.com/tmp",
+            "code": "1234"
+          },
+          "smartcard:1": {
+            "name": "smartcard ID1",
+            "role": "smartcard",
+            "selectable": true,
+            "init_prompt": "Insert smartcard",
+            "pin_prompt": "PIN"
+          },
+          "passkey": {
+            "name": "passkey",
+            "role": "passkey",
+            "selectable": true,
+            "init_instruction": "Insert security key",
+            "pin_request": true,
+            "pin_attempts": 8,
+            "pin_prompt": "Security key PIN",
+            "touch_instruction": "Touch security key"
+          }
+        },
+        "priority": [
+          "passkey",
+          "eidp",
+          "smartcard:1",
+          "password"
+        ]
+      }
+    }
+
+GDM replies with the selection, as explained in the preceding sections.
+
+Authentication mechanisms priority
+**********************************
+
+The priority for the authentication mechanisms set in the JSON message is a
+hardcoded value and follows the next priority:
+
+* Passkey
+* EIdP
+* Smartcard
+* Password
+
+Configuration options
+*********************
+
+PAM responder options
++++++++++++++++++++++
+
+PAM responder provides the new ``pam_json_services`` option to enable the
+aforementioned authentication method selection mechanism. This is a list of all
+PAM service files allowed to use it. Application maintainers can use it to let
+SSSD know they support the json protocol, and that SSSD should provide the
+available authentication methods using it.
+
+Prompt options
+++++++++++++++
+
+For the EIdP mechanism, another section called [prompting/eidp] is added. This
+section is similar to other prompting sections (i.e. 2fa). The list of options
+and their meaning:
+
+* init_prompt=your prompt here: the message that appears on the button in the
+  GUI. Default is: “Log In.”
+
+* link_prompt=your prompt here: the message that appears in the link dialog in
+  the GUI. Default is: “Log in online with another device.”
+
+For the smartcard mechanism, another section called [prompting/smartcard] is
+added. The list of options and their meaning:
+
+* init_prompt=your prompt here: the message that appears to connect the
+  smartcard. Default is: “Insert smartcard”
+
+* pin_prompt=your prompt here: the message that appears to enter the PIN.
+  Default is: “PIN”
+
+* pin_instruction=your instruction here: the message that appears to enter the
+  PIN in the reader. Default is: “Enter PIN in reader”
+
+For the passkey mechanism, another section called [prompting/passkey] is
+added. The list of options and their meaning:
+
+* pin_prompt=your prompt here: the message that appears to enter the PIN.
+  Default is: “Security key PIN”
```

Comment:
Should we use the same defaults prompts as SSSD ?

~~~
src/responder/pam/pam_prompting_config.c:28:#define DEFAULT_PASSKEY_PROMPT_INTERACTIVE _("Insert your Passkey device, then press ENTER.")
src/responder/pam/pam_prompting_config.c:29:#define DEFAULT_PASSKEY_PROMPT_TOUCH _("Please touch the device.")
~~~

Reply:
If possible, I would prefer to keep the same configuration options to adjust all user interaction messages. I will discuss with Andre who from the UX team can help us resolve this issue.

Reply:
Okay feel free to 'Resolve' this conversation when you would like to.

Reply:
I discussed this topic with Allan and taking into account that the user interaction is different (i.e. passkey CLI expects the user to press enter vs. passkey GUI GDM detects device insertion and changes screen) it's difficult to have common strings for prompts. We even discussed splitting strings, but GNOME style [advises against it](https://developer.gnome.org/hig/guidelines/writing-style.html#translations).

Last Friday Sumit, ab, Andre and I spoke about this topic and we agreed that the best path forward is to concentrate on the passwordless-GDM integration and leave the string unification for a later date. Thus, I'm removing this section and we'll have the fixed strings proposed by Allan in the [os-mockups](https://gitlab.gnome.org/Teams/Design/os-mockups/-/issues/220)